conan/internal/api/detect_api.py

```diff
@@ -122,6 +122,93 @@ def _get_e2k_architecture():
 
 
+def _parse_gnu_libc(ldd_output):
+    first_line = ldd_output.partition("\n")[0]
+    if any(glibc_indicator in first_line for glibc_indicator in ["GNU libc", "GLIBC"]):
+        return first_line.split()[-1].strip()
+    return None
+
+
+def _detect_gnu_libc(ldd="/usr/bin/ldd"):
+    if platform.system() != "Linux":
+        ConanOutput(scope="detect_api").warning("detect_gnu_libc() only works on Linux")
+        return None
+    try:
+        ldd_output = check_output_runner(f"{ldd} --version")
+        version = _parse_gnu_libc(ldd_output)
+        if version is None:
+            first_line = ldd_output.partition("\n")[0]
+            ConanOutput(scope="detect_api").warning(
+                f"detect_gnu_libc() did not detect glibc in the first line of output from '{ldd} --version': '{first_line}'"
+            )
+            return None
+        return version
+    except Exception as e:
+        ConanOutput(scope="detect_api").debug(
+            f"Couldn't determine the glibc version from the output of the '{ldd} --version' command {e}"
+        )
+    return None
+
+
+def _parse_musl_libc(ldd_output):
+    lines = ldd_output.splitlines()
+    if "musl libc" not in lines[0]:
+        return None
+    return lines[1].split()[-1].strip()
+
+
+def _detect_musl_libc(ldd="/usr/bin/ldd"):
+    if platform.system() != "Linux":
+        ConanOutput(scope="detect_api").warning(
+            "detect_musl_libc() only works on Linux"
+        )
+        return None
+
+    d = tempfile.mkdtemp()
+    tmp_file = os.path.join(d, "err")
+    try:
+        ldd_output = None
+        with open(tmp_file, 'w') as stderr:
+            check_output_runner(f"{ldd}", stderr=stderr, ignore_error=True)
+        ldd_output = load(tmp_file)
+        version = _parse_musl_libc(ldd_output)
+        if ldd_output is None:
+            first_line = ldd_output.partition("\n")[0]
+            ConanOutput(scope="detect_api").warning(
+                f"detect_musl_libc() did not detect musl libc in the first line of output from '{ldd}': '{first_line}'"
+            )
+            return None
+        return version
+    except Exception as e:
+        ConanOutput(scope="detect_api").debug(
+            f"Couldn't determine the musl libc version from the output of the '{ldd}' command {e}"
+        )
+    finally:
+        try:
+            os.unlink(tmp_file)
+        except OSError:
+            pass
+    return None
+
+
+def detect_libc(ldd="/usr/bin/ldd"):
+    if platform.system() != "Linux":
+        ConanOutput(scope="detect_api").warning(
+            f"detect_libc() is only supported on Linux currently"
+        )
+        return None
+    version = _detect_gnu_libc(ldd)
+    if version is not None:
+        return "gnu", version
+    version = _detect_musl_libc(ldd)
+    if version is not None:
+        return "musl", version
+    ConanOutput(scope="detect_api").warning(
+        f"Couldn't detect the libc provider and version"
+    )
+    return None
```

Comment:
Better return ``None, None``, it is easier for jinja templates to check, and not raise error if  used as ``lib, version = ...``

Reply:
Good catch, fixed!